apis/v1alpha1/httproute_types.go

```diff
@@ -179,13 +179,18 @@ type HTTPRouteRule struct {
 	//
 	// * The longest matching hostname.
 	// * The longest matching path.
-	// * The largest number of header matches
+	// * The largest number of header matches.
+	//
+	// If ties still exist across multiple Routes:
 	// * The oldest Route based on creation timestamp. For example, a Route with
 	//   a creation timestamp of "2020-09-08 01:02:03" is given precedence over
 	//   a Route with a creation timestamp of "2020-09-08 01:02:04".
 	// * The Route appearing first in alphabetical order (namespace/name) for
 	//   example, foo/bar is given precedence over foo/baz.
 	//
+	// If ties still exist within the Route that has been given precedence:
+	// * The first matching rule meeting the above criteria.
```

Comment:
I just realized that we are documenting the whole thing in the wrong place. Most of these rules will apply to non-HTTP cases as well.
Changing this shouldn't be part of this PR so no action required here.

Reply:
Yeah, I'd realized that as well but we have such minimal matching semantics on the other routes applying this guidance there would turn it into a much larger change. I'll create a follow up issue or PR to take this on.